sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobTestBase.java

```diff
@@ -1073,4 +1078,159 @@ protected static TokenCredential getTokenCredential(TestMode testMode) {
             return new MockTokenCredential();
         }
     }
+
+    protected static final class PagingTimeoutTestClient implements HttpClient {
+        private final Queue<String> responses = new java.util.LinkedList<>();
+
+        private int resourceCounter = 0;
+        private int pageCounter;
+
+        private enum PageType {
+            LIST_BLOBS, FIND_BLOBS, LIST_CONTAINERS
+        }
+
+        public PagingTimeoutTestClient() {
+        }
+
+        public PagingTimeoutTestClient addListBlobsResponses(int totalResourcesExpected, int maxResourcesPerPage,
+            boolean isHierarchical) {
+            return addPagedResponses(totalResourcesExpected, maxResourcesPerPage, PageType.LIST_BLOBS, isHierarchical);
+        }
+
+        public PagingTimeoutTestClient addFindBlobsResponses(int totalResourcesExpected, int maxResourcesPerPage) {
+            return addPagedResponses(totalResourcesExpected, maxResourcesPerPage, PageType.FIND_BLOBS, false);
+        }
+
+        public PagingTimeoutTestClient addListContainersResponses(int totalResourcesExpected, int maxResourcesPerPage) {
+            return addPagedResponses(totalResourcesExpected, maxResourcesPerPage, PageType.LIST_CONTAINERS, false);
+        }
+
+        private PagingTimeoutTestClient addPagedResponses(int totalResourcesExpected, int maxResourcesPerPage,
+            PageType pageType, boolean isHierarchical) {
+            int totalPagesExpected = (int) Math.ceil((double) totalResourcesExpected / maxResourcesPerPage);
+
+            for (pageCounter = 0; pageCounter < totalPagesExpected; pageCounter++) {
```

Comment:
It's a little quirky that we'd be setting the `pageCounter` and resetting it even though it's an instance property. In general, the more common pattern would be to expose a parameter on the method that needs to know what page we are on and provide it as part of the for loop.

For here, it would probably make sense to expose a boolean `isLastPage` for the `buildXmlPage` method.